internal/toproto6/function.go

```diff
@@ -30,32 +29,39 @@ func Function(ctx context.Context, fw function.Definition) *tfprotov6.Function {
 		proto.DescriptionKind = tfprotov6.StringKindPlain
 	}
 
-	for _, fwParameter := range fw.Parameters {
-		proto.Parameters = append(proto.Parameters, FunctionParameter(ctx, fwParameter))
+	for i, fwParameter := range fw.Parameters {
+		proto.Parameters = append(proto.Parameters, FunctionParameter(ctx, fw, fwParameter, i))
+	}
+
+	if fw.VariadicParameter != nil {
+		proto.VariadicParameter = FunctionParameter(ctx, fw, fw.VariadicParameter, len(fw.Parameters)+1)
 	}
 
 	return proto
 }
 
 // FunctionParameter returns the *tfprotov6.FunctionParameter for a
 // function.Parameter.
-func FunctionParameter(ctx context.Context, fw function.Parameter) *tfprotov6.FunctionParameter {
-	if fw == nil {
+func FunctionParameter(ctx context.Context, def function.Definition, param function.Parameter, position int) *tfprotov6.FunctionParameter {
+	if param == nil {
 		return nil
 	}
 
+	// TODO: what should we do with the diags? This should never happen
```

Comment:
Open question: I'm not really sure how best to handle a `diag` here without fully refactoring all of this logic (which we can do)

Same question applies to `internal/toproto5/function.go`

Reply:
Newest changes avoid diags by duplicating the logic ⚡